liblangutil/SourceReferenceExtractor.cpp

```diff
@@ -30,7 +30,7 @@ using namespace solidity::langutil;
 SourceReferenceExtractor::Message SourceReferenceExtractor::extract(
 	CharStreamProvider const& _charStreamProvider,
 	util::Exception const& _exception,
-	string _severity
+	Error::Type _type
```

Comment:
Boolean flags like `jsonOutput` are not great for readability and I wonder if we might be able to solve that better. What do you think about changing it to:
```suggestion
	variant<Error::Type, Error::Severity> _typeOrSeverity;
```
and then having separate overloads for each type: `printExceptionInformation(util::Exception const&, Error::Type)` and `printExceptionInformation(util::Exception const&, Error::Severity)`.

What do you think?

Reply:
At this point, all the locations calling the function `printExceptionInformation` send in the full `Error::Type` . At the moment, only one case requires printing of the truncated `Error::Severity`. 

If we change it to `variant < ...`  , we'll have to propagate that change down to quite a few other functions. For eg. `printExceptionInformation ( Msg )`, `formatErrorWithException()` etc . 

 Leads to unnecessary code redundancy.

Reply:
I mean, I think it would not be *that* much in terms of extra code and having the overloads would improve readability.

But it's just a suggestion. I leave that up to your judgement. If you prefer it to stay as is, that's fine too.

Reply:
Hmm.. not sure, because I just scrolling down here.. I was wondering whether it could make sense to have both at the same time.. e.g. I saw in the top `Type::Warning` or `Type::Info` - I thought that this will be just removed and managed by severity.. if we have both at the same time we could just "warn" about specific type errors - but somehow it looks like that these errors are always fatal. So maybe it makes no sense to be able to treat some of those errors as just as a "warning" or "info".

Reply:
wouldn't it not make more sense to have "severity" and something like a "message" - where both define be used to define error, warning or info "messages", where error messages are always fatal or something.. I was just thinking about this now - I didn't checked the whole context - sorry if it makes no sense at all :D

Reply:
I have refactored the code to work around `std::variant<Error::Type, Error::Severity>`  in a separate commit. It is failing some test cases right now. Will investigate that further and fix it.

Reply:
Well, it's not great but I think we can make it better - see my suggestions. With that it should actually be pretty concise and not require the extra bool/enum.

> wouldn't it not make more sense to have "severity" and something like a "message" - where both are used to define error, warning or info "messages", where error messages are always fatal or something.

@aarlt It would make sense but like I said in https://github.com/ethereum/solidity/pull/13162#discussion_r951385781, I don't think we should get into that in this PR. It would be better off as a separate change and overall it's a very low priority thing so best if we leave it for later.